hal/src/sercom/v2/i2c/flags.rs

```diff
@@ -0,0 +1,87 @@
+//! Flag definitions
+#![allow(clippy::identity_op)]
+#![allow(unused_braces)]
+
+use bitflags::bitflags;
+use core::convert::TryFrom;
+use modular_bitfield::specifiers::{B1, B5};
+use modular_bitfield::*;
+
+bitflags! {
+    /// Interrupt bitflags for I2C transactions
+    ///
+    /// The available interrupt flags are `MB`, `SB`, and `ERROR`. The binary format of the underlying bits exactly matches the
+    /// INTFLAG bits.
+    pub struct Flags: u8 {
+        /// Master on bus interrupt
+        const MB = 0x01;
+        /// Slave on bus interrupt
+        const SB = 0x02;
+        /// Error interrupt
+        const ERROR = 0x80;
+    }
+}
+
+/// Type representing the current bus state
+#[derive(BitfieldSpecifier, PartialEq)]
+pub enum BusState {
+    Unknown = 0x00,
+    Idle = 0x01,
+    Owner = 0x02,
+    Busy = 0x03,
+}
+
+/// Status flags for I2C transactions
+///
+/// The available status flags are `BUSERR`, `ARBLOST`, `RXNACK`,
+/// `BUSSTATE`, `LOWTOUT` and `CLKHOLD`, `MEXTTOUT`, `SEXTTOUT` and
+/// `LENERR`. The binary format of the underlying bits exactly matches
+/// the STATUS bits.
+#[bitfield]
+#[repr(u16)]
+pub struct Status {
+    pub buserr: bool,
+    pub arblost: bool,
+    #[skip(setters)]
+    pub rxnack: bool,
+    #[skip]
+    _reserved: B1,
+    pub busstate: BusState,
+    pub lowtout: bool,
+    #[skip(setters)]
+    pub clkhold: bool,
+    pub mexttout: bool,
+    pub sexttout: bool,
+    pub lenerr: bool,
+    #[skip]
+    _reserved: B5,
+}
+
+/// Errors available for I2C transactions
+#[derive(Debug, Clone, Copy)]
+pub enum Error {
+    BusError,
+    ArbitrationLost,
+    LengthError,
+    Nack,
+    Timeout,
+}
+
+impl TryFrom<Status> for () {
```

Comment:
I think I use this approach in the `spi` module as well, but the more I see it the more I think it's awkward and not a good solution.

I think I do like the `Result<(), Error>` pattern, but not as a `TryFrom` implementation. I think an inherent method probably works better. Thoughts?

Reply:
I agree, the impl is a little awkward. I made it an inherent method named `try_error`, feel free to suggest another name if you prefer.

Reply:
I don't love `try_error`. The `try_` prefix doesn't feel quite right, but I can't put my finger on why. Maybe `check_error` or `check_status`? It would be like to `check_bus_status`, which has a similar signature.

Reply:
Fair enough, I called it `check_bus_error`.

Reply:
Actually, what exactly is the difference between `check_bus_status` and `check_bus_error`?

Reply:
`check_bus_status` is meant to be used before starting an I2C transfer, to check that the bus is ready for use. We don't want, for example, a device to be stalling the bus. `check_bus_error` is meant to be used after finishing a transfer, to check for example is a device has NACKed, or another error has occured.